packages/protocol-http/src/extensions/http.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpHandler } from "../httpHandler";
+import { HttpHandlerExtensionConfiguration } from "./httpExtensionConfiguration";
+
+/**
+ * @internal
+ */
+export const getHttpHandlerConfiguration = <HandlerConfig extends object = Record<string, unknown>>(
+  runtimeConfig: Partial<{ httpHandler: HttpHandler<HandlerConfig> }>
+) => {
+  let httpHandler = runtimeConfig.httpHandler!;
+  const httpHandlerConfigs: HandlerConfig = httpHandler.httpHandlerConfigs();
+  return {
+    setHttpHandler(handler: HttpHandler<HandlerConfig>): void {
+      httpHandler = handler;
+    },
+    httpHandler(): HttpHandler<HandlerConfig> {
+      return httpHandler;
+    },
+    updateHttpClientConfig(key: keyof HandlerConfig, value: HandlerConfig[typeof key]): void {
+      httpHandlerConfigs[key] = value;
+    },
+    httpHandlerConfigs(): HandlerConfig {
+      return httpHandlerConfigs;
```

Comment:
this doesn't look right, the value won't change since it is a const ref. 

should it be?:
```
return this.httpHandler().httpHandlerConfigs()
```